src/protocols/interaction_model/StatusCode.h

```diff
@@ -93,13 +93,13 @@ class StatusCode
      *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum)
      * @param cluster_specific_code - cluster-specific code to record with the failure
      *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum::kWindowNotOpen)
-     * @return a StatusCode instance properly configured.
+     * @return a ClusterStatus instance properly configured.
      */
     template <typename T>
-    static StatusCode ClusterSpecificFailure(T cluster_specific_code)
+    static ClusterStatus ClusterSpecificFailure(T cluster_specific_code)
     {
         static_assert(std::numeric_limits<T>::max() <= std::numeric_limits<uint8_t>::max(), "Type used must fit in uint8_t");
-        return StatusCode(Status::Failure, static_cast<uint8_t>(cluster_specific_code));
+        return ClusterStatus(Status::Failure, chip::to_underlying(cluster_specific_code));
```

Comment:
```suggestion
        return ClusterStatus(Status::Failure, to_underlying(cluster_specific_code));
```

We're inside `namespace chip { }` here.

Reply:
Leaving `chip::to_underlying` since it's not the same as `std::to_underlying` and I really want to make it clear that it's our special version.